controllers/component_dependency_update_controller_test.go

```diff
@@ -50,25 +52,46 @@ const (
-	if failures == 0 {
-		return DefaultTestDependenciesUpdate(ctx, client, scheme, eventRecorder, downstreamComponents, result)
+type TestComponentDependenciesUpdater struct {
+	componentDependenciesUpdater renovate.ComponentDependenciesUpdater
+	eventRecorder                record.EventRecorder
+	failures                     int
+}
+
+func NewTestComponentDependenciesUpdater(client client.Client, scheme *runtime.Scheme, eventRecorder record.EventRecorder) *TestComponentDependenciesUpdater {
+	return &TestComponentDependenciesUpdater{
+		componentDependenciesUpdater: renovate.NewDefaultComponentDependenciesUpdater(client, scheme, eventRecorder),
+		eventRecorder:                eventRecorder,
+		failures:                     0,
 	}
-	failures = failures - 1
-	return true, fmt.Errorf("failure")
 }
 
-func DefaultTestDependenciesUpdate(ctx context.Context, client client.Client, scheme *runtime.Scheme, eventRecorder record.EventRecorder, downstreamComponents []applicationapi.Component, result *BuildResult) (immediateRetry bool, err error) {
-	log := ctrllog.FromContext(ctx)
-	for _, downstreamComponent := range downstreamComponents {
-		log.Info(fmt.Sprintf("Nudging %s due to successful build of %s", downstreamComponent.Name, result.Component.Name), l.Action, l.ActionUpdate)
-		//TODO: do we want the event? It's just for unit testing at the moment
-		eventRecorder.Event(&downstreamComponent, v1.EventTypeNormal, ComponentNudgedEventType, fmt.Sprintf("component %s.%s was nudged by successful build of %s that produces image %s:%s@%s", downstreamComponent.Namespace, downstreamComponent.Name, result.Component.Name, result.BuiltImageRepository, result.BuiltImageTag, result.Digest))
+func (t *TestComponentDependenciesUpdater) Update(ctx context.Context, components []*git.ScmComponent, buildResult *renovate.BuildResult) error {
+	if t.failures == 0 {
+		for _, downstreamComponent := range components {
+			log.Info("Nudging due to successful build",
+				"component", downstreamComponent.ComponentName(),
+				"updatedcomponent", buildResult.UpdatedComponentName,
+				l.Action, l.ActionUpdate)
+			//TODO: do we want the event? It's just for unit testing at the moment
+			component := &applicationapi.Component{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      downstreamComponent.ComponentName(),
+					Namespace: downstreamComponent.NamespaceName(),
+				},
+			}
+			t.eventRecorder.Event(component, v1.EventTypeNormal, ComponentNudgedEventType, fmt.Sprintf("component %s.%s was nudged by successful build of %s that produces image %s:%s@%s", downstreamComponent.NamespaceName(), downstreamComponent.ComponentName(), buildResult.UpdatedComponentName, buildResult.BuiltImageRepository, buildResult.BuiltImageTag, buildResult.Digest))
 
+		}
+		if err := t.componentDependenciesUpdater.Update(ctx, components, buildResult); err != nil {
+			return err
+		}
+		return nil
 	}
-	return false, nil
+	t.failures = t.failures - 1
```

Comment:
Why do we decrease failures count? Or is it a remainder? Please make it clear via name or comment.